src/api/route-services/pipelines.js

```diff
@@ -0,0 +1,196 @@
+const AWS = require('aws-sdk');
+const crypto = require('crypto');
+const fetch = require('node-fetch');
+const YAML = require('yaml');
+const jq = require('jq-web');
+const config = require('../../config');
+
+
+class PipelinesService {
+  constructor() {
+    this.stepFunctions = new AWS.StepFunctions({
+      region: config.awsRegion,
+    });
+
+    this.clusterInfo = null;
+  }
+
+  async getClusterInformation() {
+    if (!this.clusterInfo) {
+      const eks = new AWS.EKS({
+        region: config.awsRegion,
+      });
+
+      const accountId = await config.awsAccountIdPromise();
+      const info = await eks.describeCluster({ name: `biomage-${config.clusterEnv}` }).promise();
+
+      const { name, endpoint } = info.cluster;
+      const certAuthority = info.cluster.certificateAuthority.data;
+      const roleArn = `arn:aws:iam::${accountId}:role/state-machine-role-${config.clusterEnv}`;
+
+      this.clusterInfo = {
+        accountId, name, endpoint, certAuthority, roleArn,
+      };
+    }
+
+    return this.clusterInfo;
+  }
+
+  static async getPipelineImage() {
+    const response = await fetch(
+      config.pipelineInstanceConfigUrl,
+      {
+        method: 'GET',
+      },
+    );
+
+    const txt = await response.text();
+    const manifest = YAML.parseAllDocuments(txt);
+
+    return jq.json(manifest, '..|objects|.r//empty');
+  }
+
+  async generatePipeline(experimentId) {
+    const { name, endpoint, certAuthority } = await this.getClusterInformation();
+    const namespace = `worker-${config.sandboxId}`;
```

Comment:
this should probably live in the config

Reply:
I was thinking about that, but it is only used once, and it's directly derived from another config variable. If you feel like it's a better place for it, I'll do it though.

Reply:
I think it is better to move it. In the future, if we decided to change the naming conventions for the namespace, we could forget about this line here and break the code, because of that I think it is better to have it all in one place.